### ISSUE TITLE
document is not defined (SvelteKit)

### ISSUE THREAD

**Issue by ItsBioDude** [f5/unovis#202]
So here is the issue
```powershell
ReferenceError: document is not defined
    at new ComponentCore (file:///D:/Hobby/thuchanh/node_modules/.pnpm/@unovis+ts@1.1.1/node_modules/@unovis/ts/core/component/index.js:25:28)
    at new XYComponentCore (file:///D:/Hobby/thuchanh/node_modules/.pnpm/@unovis+ts@1.1.1/node_modules/@unovis/ts/core/xy-component/index.js:9:9)
    at new Line (file:///D:/Hobby/thuchanh/node_modules/.pnpm/@unovis+ts@1.1.1/node_modules/@unovis/ts/components/line/index.js:16:9)
    at eval (/node_modules/.pnpm/@unovis+svelte@1.1.1_@unovis+ts@1.1.1_svelte@3.59.1/node_modules/@unovis/svelte/components/line/line.svelte:26:20)
    at Object.$$render (/node_modules/.pnpm/svelte@3.59.1/node_modules/svelte/internal/index.mjs:2042:22)  
    at Object.default (/src/routes/tara/+page.svelte:135:100)
    at eval (/node_modules/.pnpm/@unovis+svelte@1.1.1_@unovis+ts@1.1.1_svelte@3.59.1/node_modules/@unovis/svelte/containers/xy-container/xy-container.svelte:72:156)
    at Object.$$render (/node_modules/.pnpm/svelte@3.59.1/node_modules/svelte/internal/index.mjs:2042:22)  
    at Object.default (/src/routes/tara/+page.svelte:133:103)
    at eval (/src/lib/StatisticPage.svelte:22:33)
    at Object.$$render (/node_modules/.pnpm/svelte@3.59.1/node_modules/svelte/internal/index.mjs:2042:22)  
    at eval (/src/routes/tara/+page.svelte:114:170)
    at Object.$$render (/node_modules/.pnpm/svelte@3.59.1/node_modules/svelte/internal/index.mjs:2042:22)  
    at Object.default (/.svelte-kit/generated/root.svelte:74:138)
    at eval (/node_modules/.pnpm/@sveltejs+kit@1.16.3_svelte@3.59.1_vite@4.3.5/node_modules/@sveltejs/kit/src/runtime/components/layout.svelte:8:41)
    at Object.$$render (/node_modules/.pnpm/svelte@3.59.1/node_modules/svelte/internal/index.mjs:2042:22)  
    at eval (/.svelte-kit/generated/root.svelte:63:129)
    at $$render (/node_modules/.pnpm/svelte@3.59.1/node_modules/svelte/internal/index.mjs:2042:22)
    at Object.render (/node_modules/.pnpm/svelte@3.59.1/node_modules/svelte/internal/index.mjs:2050:26)    
    at Module.render_response (/node_modules/.pnpm/@sveltejs+kit@1.16.3_svelte@3.59.1_vite@4.3.5/node_modules/@sveltejs/kit/src/runtime/server/page/render.js:185:29)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Module.render_page (/node_modules/.pnpm/@sveltejs+kit@1.16.3_svelte@3.59.1_vite@4.3.5/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:294:10)
    at async resolve (/node_modules/.pnpm/@sveltejs+kit@1.16.3_svelte@3.59.1_vite@4.3.5/node_modules/@sveltejs/kit/src/runtime/server/respond.js:398:17)
    at async Module.respond (/node_modules/.pnpm/@sveltejs+kit@1.16.3_svelte@3.59.1_vite@4.3.5/node_modules/@sveltejs/kit/src/runtime/server/respond.js:267:20)
    at async file:///D:/Hobby/thuchanh/node_modules/.pnpm/@sveltejs+kit@1.16.3_svelte@3.59.1_vite@4.3.5/node_modules/@sveltejs/kit/src/exports/vite/dev/index.js:506:22
```
And here is my code:

```svelte
<script>
	import 'virtual:uno.css';
	import { presetUno } from 'unocss';
	import { createForm } from 'felte';
	import { DateInput } from 'date-picker-svelte';

	// TODO: Check back on this (uninstall?)
	// import Line from 'svelte-chartjs/Line.svelte';
	import {
		VisXYContainer,
		VisLine,
		VisAxis,
		VisBulletLegend,
	} from '@unovis/svelte';

	import StatisticPage from '$lib/StatisticPage.svelte';
	import DateCard from '$lib/DateCard.svelte';
	import { supabase } from '$lib/supabaseClient';
	import OpenFormButton from '$lib/OpenFormButton.svelte';
	import SubmitForm from '$lib/SubmitForm.svelte';
	import dayjs from 'dayjs';

	// Query Data from Supabase
	let visible = false;
	let page = 0;
	let size = 10;
	let created_at = new Date();
	// let dates = [];
	export let data;


	let { dates } = data;
	$: ({ dates } = data);
	dates.sort((a, b) => a.id - b.id); // b - a for reverse sort
	console.log(dates);
	let dates_length = dates.length;


	let lineGraphDates = dates.slice(-10);
	console.log(lineGraphDates);

	let DataRecord = [];
	lineGraphDates.forEach((e, i) => {
		DataRecord.push({
			x: i,
			// x: e.created_at,
			y: e.tara,
		});
	});

	console.log(DataRecord);
// Output: 
/*
[
    {
        "x": 0,
        "y": 10
    },
    {
        "x": 1,
        "y": 14
    },
    {
        "x": 2,
        "y": 13
    },
    {
        "x": 3,
        "y": 0
    },
    {
        "x": 4,
        "y": 10
    },
    {
        "x": 5,
        "y": 1
    },
    {
        "x": 6,
        "y": 7
    },
    {
        "x": 7,
        "y": 10
    },
    {
        "x": 8,
        "y": 10
    },
    {
        "x": 9,
        "y": 10
    }
]
*/
</script>

			<VisXYContainer data={DataRecord}>
				<VisLine data={DataRecord} x={(d) => d.x} y={(d) => d.y} />
				<VisAxis type="x" />
				<VisAxis type="y" />
			</VisXYContainer>
```
Sorry for my bad english but any helps, sometimes it works but sometimes doesn't and it output error like above

**Comment by ItsBioDude** [comment]
Tbh this is weird, the way to workaround is:

First comment out the ```<VisXYContainer data={DataRecord}>...</VisXYContainer>``` then ```pnpm run dev``` after that uncomment ```<VisXYContainer data={DataRecord}>...</VisXYContainer>``` and everything works

**Comment by reb-dev** [comment]
@ItsBioDude thanks for reporting! I'll take a look and see if I can resolve it.

**Comment by ItsBioDude** [comment]
Uhm I guess this is just a small issue with svelte. So it seem like svelte want to run the code outside the browser which return 500 - error code. So the fix for this is simple (just make sure the code only run on browser)

```svelte
<script>
	import { browser } from '$app/environment';
// The rest set up for unovis
</script>

		{#if browser}
			<StatisticLineGraph
				{bulletLegendItems}
				{DataRecord}
				{axisY}
				{lineGraphDates}
			/>
			<!-- TODO: Fix the graph -->{/if}
```

I accidentally read abt this on [a post about making theme toggler on dev.to](https://dev.to/willkre/persistent-theme-switch-dark-mode-with-svelte-sveltekit-tailwind-1b9g). It would be nice if you could may be add this change to docs to let other people knows as well. Thanks

**Comment by rokotyan** [comment]
@mankins @ItsBioDude We've just released Unovis 1.2 that should take care of that automatically. Please let us know if you encounter any problems with it.
